pkg/yurthub/proxy/util/util.go

```diff
@@ -295,3 +300,31 @@ func WithRequestTimeout(handler http.Handler) http.Handler {
 		handler.ServeHTTP(w, req)
 	})
 }
+
+func WithSaTokenSubstitute(handler http.Handler, tenantMgr tenant.Interface) http.Handler {
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+
+		if tenantMgr.GetTenantToken() == "" {
+			klog.V(2).Info("tenant token is empty.")
+		} else if oldToken := util.ParseBearerToken(req.Header.Get("Authorization")); oldToken != "" { //bearer token is not empty&valid
+
+			if jsonWebToken, err := jwt.ParseSigned(oldToken); err != nil {
+
+				klog.Errorf("invaled bearer token %s, err: %v", oldToken, err)
+			} else {
+				oldClaim := jwt2.BearerClaims{}
```

Comment:
in order to get `namespace` from token, we can only use subject in `jwt.Claims` instead of defining `BearerClaims`. we can reference code of the subject format here: 
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/serviceaccount/tokens_controller.go#L404
https://github.com/kubernetes/kubernetes/blob/master/pkg/serviceaccount/legacy.go#L32-L41

Reply:
```
type legacyPrivateClaims struct {
	ServiceAccountName string `json:"kubernetes.io/serviceaccount/service-account.name"`
	ServiceAccountUID  string `json:"kubernetes.io/serviceaccount/service-account.uid"`
	SecretName         string `json:"kubernetes.io/serviceaccount/secret.name"`
	Namespace          string `json:"kubernetes.io/serviceaccount/namespace"`
}
```

legacyPrivateClaims  is private, so I copy the type to a new type.

Reply:
@MrGirl we only need namespace, so `jwt.Claims` can be used. we need try our best to reduce the code complexity.

Reply:
there's no namespace field in jwt.Claims.  I this this is the reason why kubernetes had to define the type legacyPrivateClaims with a Namespace filed.

Reply:
we can get namespace from `jwt.Claims.Subject` field. the contents of `Subject` is `apiserverserviceaccount.MakeUsername(serviceAccount.Namespace, serviceAccount.Name)`.